lale/operators.py

```diff
@@ -3684,6 +3684,65 @@ def _final_individual_op(self) -> Optional["IndividualOp"]:
         else:
             return op._final_individual_op
 
+    def replace(
+        self, original_op: Operator, replacement_op: Operator
+    ) -> "BasePipeline[OpType]":
+        """Replaces an original operator with a replacement operator for the pipeline.
+
+        Parameters
+        ----------
+        original_op :
+            Operator to replace within pipeline.
+
+        replacement_op :
+            Operator to replace the original with.
+
+        Returns
+        -------
+        modified_pipeline :
+            Modified pipeline where original operator is replaced with replacement.
+        """
+
+        new_steps: List[Operator] = []
+        new_edges: List[Tuple[Operator, Operator]] = []
+
+        replacements: Dict[Operator, Operator] = {}
+
+        # NOTE: needs to deal with multiple levels of nesting
+        for step in self.steps():
+            if step == original_op:
+                new_steps.append(replacement_op)
+            elif isinstance(step, OperatorChoice):
+                old_choice = step
+                new_choice = step
+                if original_op in step.steps():
+                    old_choice = step
+                    if step.steps().index(original_op) == 0:
```

Comment:
The code here has an implicit assumption that a choice has exactly two elements. Can you please generalize this for n-ary choices?